tracing-subscriber/src/fmt/format/json.rs

```diff
@@ -236,6 +236,28 @@ where
                 serializer.serialize_entry("target", meta.target())?;
             }
 
+            if self.display_thread_name {
+                match std::thread::current().name() {
+                    Some(name) => {
+                        serializer.serialize_entry("threadName", name)?;
+                    }
+                    None => {
+                        // fall-back to thread id when name is absent and id's are not enabled
+                        if !self.display_thread_id {
+                            serializer.serialize_entry(
+                                "threadId",
+                                &format!("{:?}", std::thread::current().id()),
+                            )?;
+                        }
+                    }
```

Comment:
nit/style: take it or leave it, but we could consider collapsing the `if` into the match arm here:
```suggestion
                    // fall-back to thread id when name is absent and id's are not enabled
                    None if !self.display_thread_id => {
                        serializer.serialize_entry(
                            "threadId",
                            &format!("{:?}", std::thread::current().id()),
                        )?;
                    },
                    _ => {}
```